src/pybind/mgr/restful/module.py

```diff
@@ -599,6 +602,7 @@ def submit_request(self, _request, **kwargs):
         with self.requests_lock:
             request = CommandsRequest(_request)
             self.requests.append(request)
+            self.requests = self.requests[-self.max_requests:]
```

Comment:
i'd like to see the analysis of the impact of this change.

Reply:
I ran few thousand pg dump command with and without the trimming of the requests array, there is a big improvement with the trimming, users that already checked that fix as patch mentioned that it helped a bit, but the manger was still consuming more memory and finally OOM and killed.

with the trimming:
![image](https://github.com/ceph/ceph/assets/97529641/29267161-d966-4646-9182-7d8f679a1cbe)


without the trimming:
![image](https://github.com/ceph/ceph/assets/97529641/3437ca43-e84a-4817-8876-0bf6ee6677a5)

Reply:
@NitzanMordhai could you please extract this commit into another PR? the massif output is great. but it is not analysis. by analysis, i mean, the pros and cons of having an unbounded trim limit. and explain the reason why we need to keep `self.requests` to reviewers.

Reply:
I think I have the same question as @tchaikov -- as a reviewer, I don't actually know what the purpose of self.requests is.  If it's safe to trim it, do we need it at all?  Why is it mandatory to maintain 500 of them, but not 501?